### PR TITLE
PrivGetFilterLogs needs to check privacy group membership at given block

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

When getting private filter logs, pass block number through so group membership can be checked at that point in time.

See #882

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java [hunk 1]
```diff
@@ -266,6 +266,16 @@ void recordPendingTransactionEvent(final Transaction transaction) {
     return logs;
   }
 
+  public BlockParameter getToBlock(final String filterId) {
+    final LogFilter filter = filterRepository.getFilter(filterId, LogFilter.class).orElse(null);
+    if (filter == null) {
+      return null;
+    } else {
+      filter.resetExpireTime();
```
Comment (lucassaldanha) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java:274]: Why do we need to reset the expire time here?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/PrivGetFilterChanges.java [hunk 2]
```diff
@@ -53,7 +57,8 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
     final String privacyGroupId = requestContext.getRequiredParameter(0, String.class);
     final String filterId = requestContext.getRequiredParameter(1, String.class);
 
-    checkIfPrivacyGroupMatchesAuthenticatedEnclaveKey(requestContext, privacyGroupId);
+    final BlockParameter blockParameter = filterManager.getToBlock(filterId);
```
Comment (lucassaldanha) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/PrivGetFilterChanges.java:60]: If `filterManager.getToBlock` can return null, we need to be careful when using the `blockParameter` variable in `getBlockNumberToCheckForGroupMembership `

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/PrivGetFilterLogs.java [hunk 3]
```diff
@@ -53,7 +55,8 @@ public JsonRpcResponse response(final JsonRpcRequestContext request) {
     final String privacyGroupId = request.getRequiredParameter(0, String.class);
     final String filterId = request.getRequiredParameter(1, String.class);
 
-    checkIfPrivacyGroupMatchesAuthenticatedEnclaveKey(request, privacyGroupId);
+    final BlockParameter blockParameter = filterManager.getToBlock(filterId);
```
Comment (lucassaldanha) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/PrivGetFilterLogs.java:58]: Same thing about `blockParameter` being null.

Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/PrivGetFilterLogs.java:58]: fixed

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/PrivGetFilterChanges.java (hunk 2 above)
Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/PrivGetFilterChanges.java:60]: fixed

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java (hunk 1 above)
Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java]: removed

**Comment by macfarla** [comment]
Assuming filter will be removed when user is removed from the privacy group, there's nothing to do here.

**Pull request closed without merging**